crates/binjs_io/src/io.rs

```diff
@@ -93,8 +93,32 @@ pub trait TokenReader where Self::Error: Debug + From<::TokenReaderError>,
     /// read (in particular that all bytes were consumed). In most
     /// implementations, failure to do so will raise an assertion.
     fn untagged_tuple(&mut self) -> Result<Self::UntaggedGuard, Self::Error>;
+
+    // Functions for file format dump.
```

Comment:
What's a dump, exactly?

Reply:
Also, it looks like it's going to make the implementation of this trait more complicated. I suspect that providing default noop implementations would be useful, what do you think?